wp-content/themes/chrisdinh-photography/resources/views/components/slider.blade.php

```diff
@@ -1,27 +1,53 @@
+<!--front-end view of the slider component-->
+<!--run yarn build/dev before coding to implement new tailwind code-->
 @unless ( empty($data) )
-    <section id="{{ $sliderSettings['slider__id'] }}"  class="slider splide relative block w-screen h-screen"
+    <!--parent container of the slider -->
+    <!--uses splidejs.com, read the documentation-->
+    <section id="{{ $sliderSettings['slider__id'] }}"  class="relative w-screen h-screen splider splide"
         data-type="{{ $sliderSettings['slider__type'] }}"
         data-speed="{{ implode(',', $sliderSettings['slider__speed'])  }}"
         data-width="{{ implode(',', $sliderSettings['slider__width']) }}"
         data-height="{{ implode(',', $sliderSettings['slider__height']) }}"
         data-per-page="{{ implode(',', $sliderSettings['slider__per-page']) }}"
         data-per-move="{{ implode(',', $sliderSettings['slider__per-move']) }}"
         data-gap="{{ implode(',', $sliderSettings['slider__gap']) }}"
-        data-arrows="{{ implode(',', $sliderSettings['slider__arrows']) }}"
-        data-pagination="{{ implode(',', $sliderSettings['slider__pagination']) }}"
+        @if (get_field('slider__show-arrows'))
```

Comment:
You don't need to run a `get_field` function in your view template. This costs an additional database query which we already doing in our Composer file. The data that gets passed to the template is handled in `app/View/Components` or `app/View/Composers` to keep the view template as clean as possible. `$sliderSettings` is a variable that exists in the view already and contains your field. In the file `app/View/Composers/FrontPage.php`, look at the following method:

```php
private function getHomepageSliderSettings() {
        return get_field('homepage__slider-settings');
    }
```

This method fetches all fields inside the slider setting group - so as long as you are adding a slider setting to that group, it will be available in the variable `$sliderSettings`. You can replace this conditional with the following and it should work (not tested but it should work)

```php
@if ($sliderSettings['slider__show-arrows'])
...
@endif
```

Reply:
updated in commit 7a8dd48d77f9683e02904cbb06e5dc28970869e9